spec/ntheory_spec.rb

```diff
@@ -0,0 +1,126 @@
+require 'spec_helper'
+
+describe SymEngine do
+
+  before :each do
+  end
+
+  describe 'NTheory' do
+    before :each do
+      @i1 = SymEngine::Integer.new(1)
+      @i4 = SymEngine::Integer.new(4)
+      @i5 = SymEngine::Integer.new(5)
+      @im4 = SymEngine::Integer.new(-4)
+      @im5 = SymEngine::Integer.new(-5)
+    end
+
+    describe '#gcd' do
+      context 'GCD of 2 and 4' do
+        it 'returns 2' do
+          f = SymEngine::gcd(2, 4)
+          expect(f).to eql(2)
+        end
+      end
+    end
+
+    describe '#lcm' do
+      context 'LCM of 2 and 4' do
+        it 'returns 4' do
+          f = SymEngine::lcm(2, 4)
+          expect(f).to eql(4)
+        end
+      end
+    end
+
+    describe '#nextprime' do
+      context 'NextPrime after 4' do
+        it 'returns 5' do
+          f = SymEngine::nextprime(4)
+          expect(f).to eql(5)
+        end
+      end
+    end
+
+    describe '#mod' do
+      context '5 mod 4' do
+        it 'returns 1' do
+          f = @i5 % @i4
+          expect(f).to eql(1)
+        end
+      end
+      context '-5 mod -4' do
+        it 'returns -1' do
+          f = @im5 % @im4
+          expect(f).to eql(-1)
+        end
+      end
+      context '5 mod -4' do
+        it 'returns -3' do
+          f = @i5 % @im4
+          expect(f).to eql(-3)
+        end
+      end
+      context '-5 mod 4' do
+        it 'returns 3' do
+          f = @im5 % @i4
+          expect(f).to eql(3)
```

Comment:
So, `f` is a `SymEngine::Integer`, right? Will implicit conversion happen here while comparing?